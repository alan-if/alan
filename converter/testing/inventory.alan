OBJECT ATTRIBUTES
    takeable.

CONTAINER inventory
    LIMITS
	count 3 THEN "You can't pick up anything more."

    HEADER
	"You are carrying"

    ELSE
	"You are empty-handed."
END CONTAINER inventory.

VERB take, get, pick
    CHECK OBJECT HERE
        ELSE "I can't see any $o here!"
    AND OBJECT NOT IN inventory
	ELSE "You already have the $o."
    AND OBJECT IS takeable
	ELSE "You can't take the $o."
    DOES
        LOCATE OBJECT IN inventory.
        "Taken."
END VERB.


VERB drop
    CHECK OBJECT IN INVENTORY
	ELSE "You don't have the $o!"
    DOES
        LOCATE OBJECT HERE.
	"Dropped."
END VERB.


SYNTAX i = i.
SYNONYMS invent = i.
VERB i
    DOES
	LIST inventory.
END VERB.


LOCATION loc
END LOCATION.

OBJECT o1 At loc
ARTICLE "an"
END OBJECT.

OBJECT o2 At loc
ARTICLE "an"
END OBJECT.

OBJECT o3 At loc
ARTICLE "an"
END OBJECT.

OBJECT o4 At loc
ARTICLE "an"
END OBJECT.


START AT loc.
