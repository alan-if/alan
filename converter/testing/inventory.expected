########## inventory ##########

Add To Every object
  Is
    takeable.
End Add To.

The inventory
  Container
    Limits
      Count 3
        Else
          "You can't pick up anything more."
    Header
      "You are carrying"
    Else
      "You are empty-handed."
End The inventory.

Verb take, get, pick
  Check object Here
    Else 
      "I can't see any $o here!"
  And object Not In inventory
    Else 
      "You already have the $o."
  And object Is takeable
    Else 
      "You can't take the $o."
  Does
    Locate object In inventory.
    "Taken."
End Verb.

Verb drop
  Check object In inventory
    Else 
      "You don't have the $o!"
  Does
    Locate object Here.
    "Dropped."
End Verb.

Syntax
  i = i.

Synonyms
  invent = i.
  

Verb i
  Does
    List inventory.
End Verb.


The loc Isa location
  
End The loc.

The o1 Isa object
  At loc 
  Article
    "an"
  
End The o1.

The o2 Isa object
  At loc 
  Article
    "an"
  
End The o2.

The o3 Isa object
  At loc 
  Article
    "an"
  
End The o3.

The o4 Isa object
  At loc 
  Article
    "an"
  
End The o4.

Start At loc.
  
